__tests__/app.test.js

```diff
@@ -34,6 +34,60 @@ describe("/api/topics", () => {
 });
 
 describe("/api/articles", () => {
+  describe("GET", () => {
+    it("200: responds with an array of all articles properties including the correct author and comment_count", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length).toBe(12);
+          body.articles.forEach((article) => {
+            expect(article).toEqual(
+              expect.objectContaining({
+                author: expect.any(String),
+                title: expect.any(String),
+                article_id: expect.any(Number),
+                topic: expect.any(String),
+                created_at: expect.any(String),
+                votes: expect.any(Number),
+                comment_count: expect.any(Number),
+              })
+            );
+          });
+        });
+    });
+    it("200: should be sorted by date in Ascending order", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          for (let i = 0; i < body.articles.length - 1; i++) {
+            expect(
+              body.articles[i].created_at > body.articles[i + 1].created_at
+            ).toBe(true);
+          }
+        });
+    });
+    it("200: responds with specified topic when query is input", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length === 11).toBe(true);
+          body.articles.forEach((article) => {
+            expect(article.topic).toBe("mitch");
+          });
+        });
+    });
+    it("404: topic not found in article data", () => {
+      return request(app)
+        .get("/api/articles?topic=not_mentioned")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("not found");
+        });
+    });
+  });
```

Comment:
TEST

- [ ] 400 invalid topic
- [ ]  200 for an topic that exists but has no articles